### ISSUE TITLE
Object interface should fail for incorrectly formatted suite

### ISSUE THREAD

**Issue by jason0x43** [theintern/intern#794]
Intern 4 object interface suites should have one one of two formats, depending on whether suite lifecycle functions (before, after, beforeEach, etc.) are being specified:

```js
{
    test1() { ... },
    test2() { ... },
    test3() { ... }
}
```

or

```js
{
    before() { ... },
    tests: {
        test1() { ... },
        test2() { ... },
        test3() { ... }
    }
}
```

Currently the object interface will accept suites where tests are mixed in with lifecycle functions. It will simply ignore the tests. Instead, it should throw an error.

**Comment by jason0x43** [comment]
Correction: it doesn't ignore the tests, it treats the lifecycle methods as tests. The current logic used to identify an object suite descriptor is the presents of a 'tests' property with an object value. If a 'tests' property is present, the descriptor is assumed to have tests and lifecycle methods. If a 'tests' property is not present, everything is assumed to be a test.

**Comment by jason0x43** [comment]
This is resolved in 9f2c293. Since this isn't really an error case (the whole idea behind adding the `tests` property to the object interface was to reduce the number of reserved property names), the executor just emits log messages indicating that tests are being created with lifecycle method names. These aren't normally visible, but are displayed by the default reporter when the `debug` option is set.
